Neos.RedirectHandler.DatabaseStorage/Classes/Domain/Repository/RedirectRepository.php

```diff
@@ -185,9 +185,8 @@ public function findDistinctHosts()
     public function incrementHitCount(RedirectInterface $redirect)
     {
         /** @var Query $query */
-        $query = $this->entityManager->createQuery('UPDATE Neos\RedirectHandler\DatabaseStorage\Domain\Model\Redirect r SET r.hitCounter = r.hitCounter + 1 WHERE r.sourceUriPath = :sourceUriPath and r.host = :host');
-        $query->setParameter('host', $redirect->getHost())
-            ->setParameters([
+        $query = $this->entityManager->createQuery('UPDATE Neos\RedirectHandler\DatabaseStorage\Domain\Model\Redirect r SET r.hitCounter = r.hitCounter + 1 WHERE r.sourceUriPath = :sourceUriPath and (r.host = :host or r.host IS NULL)');
```

Comment:
Risky it can update two redirect if the same redirect exist (one with host = null and the other one with a host set).

Best solution is to the the value of ```$redirect->getHost()``` and change the query from ```host = :host``` to ```host IS NULL```